### ISSUE TITLE
Converting mxnet checkpoint to Tensorflow model

### ISSUE THREAD

**Issue by cheshirecats** [apache/incubator-mxnet#4882]
Currently the best method to run CNN in browser might be Keras.js (with Tensorflow backend):

https://github.com/transcranial/keras-js

On the other hand, mxnet is probably the fastest in training models.

So I wonder if there's any tools for converting trained mxnet checkpoints (.json & .params) to Tensorflow models. Thanks.

**Comment by zihaolucky** [comment]
Is the tool backend relevant? Or it just rely on Keras,  which means it uses the Keras output, the graph definition and params. If so, https://github.com/dmlc/keras might do the trick.

**Comment by cheshirecats** [comment]
Hi. It just needs the model to be loadable in Keras, and then it can be exported & converted for Keras.js to use (as in Keras.js README). 

Regarding https://github.com/dmlc/keras , is it a project to use mxnet as a backend for Keras?  Thanks.

**Comment by zihaolucky** [comment]
Yes, but I haven't participate in this project. @piiswrong

**Comment by cheshirecats** [comment]
@zihaolucky Hi it seems Keras.js requires the model to be a Keras model, so that it be exported using save_weights and to_json, as written in https://github.com/transcranial/keras-js .

Therefore other models need to be converted to Keras models first. It would be great if such a tool can be offered.

**Comment by yajiedesign** [comment]
This issue is closed due to lack of activity in the last 90 days. Feel free to reopen if this is still an active issue. Thanks!

**Comment by entropymaster** [comment]
I am looking to do the same. Did someone find a way out for this?

**Comment by futurely** [comment]
https://github.com/pertusa/MXNetToCaffeConverter
https://github.com/cypw/MXNet2Caffe
https://github.com/wranglerwong/Mxnet2Caffe

https://github.com/ethereon/caffe-tensorflow
https://github.com/hahnyuan/nn_tools

**Comment by szha** [comment]
@apache/mxnet-committers: This issue has been inactive for the past 90 days. It has no label and needs triage.

For general "how-to" questions, our [user forum](https://discuss.mxnet.io/) (and [Chinese version](https://discuss.gluon.ai/)) is a good place to get help.
